### PR TITLE
Feature/spark/support mongostream source

### PR DESCRIPTION
**Pull request by Imbruced**
### Problem

Add support for sending events when reading from mongo db using spark streaming.

Closes: #2886

### Solution
Mongo streaming visitor + tests.

#### One-line summary:
Mongo streaming visitor + tests.

### Checklist

- [ ] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [ ] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

File: integration/spark/app/build.gradle [hunk 1]
```diff
@@ -94,10 +94,12 @@ dependencies {
     compileOnly("org.apache.spark:spark-hive_${scala}:${spark}")
     compileOnly("org.apache.spark:spark-sql-kafka-0-10_${scala}:${spark}")
 
+
     testImplementation("org.apache.spark:spark-core_${scala}:${spark}")
     testImplementation("org.apache.spark:spark-sql_${scala}:${spark}")
     testImplementation("org.apache.spark:spark-hive_${scala}:${spark}")
     testImplementation("org.apache.spark:spark-sql-kafka-0-10_${scala}:${spark}")
+    testImplementation("org.mongodb.spark:mongo-spark-connector_${scala}:10.1.1")
```
Comment (pawel-big-lebowski) [inline integration/spark/app/build.gradle:102]: Would it make sense to have integration test in docker that downloads `mongo-spark-connector` when being run? 
I know we're doing a lot of  `testImplementation` dependencies in openlineage-spark repo, but the problem with this are: 
 * We're polluting test classpath with classes that will be present in test runtime but won't be available at the runtime. 
 * We're limiting the possibility to run tests against different mongo connector versions in the future which would be possible for docker tests. 

What do you think?

Comment (Imbruced) [reply integration/spark/app/build.gradle:102]: Make sense I can put the test in a different file and run spark in container.

**Comment by Imbruced** [comment]
@pawel-big-lebowski Spark 3.5 tests were failing due to outdated version of the connector, I bumped it :)

File: integration/spark/app/build.gradle (hunk 1 above)
Comment (pawel-big-lebowski) [reply integration/spark/app/build.gradle]: Great test that proves the feature is working. thank you

**Review by pawel-big-lebowski — APPROVED** [review]

**Comment by pawel-big-lebowski** [comment]
@Imbruced would you mind resolving conflict?

**Pull request merged**